src/RecoveryServices/RecoveryServices.Backup.Providers/Providers/AzureFilesPsBackupProvider.cs

```diff
@@ -480,28 +524,50 @@ private RestAzureNS.AzureOperationResponse<ProtectionPolicyResource> CreateorMod
+                    AzureWorkloadProviderHelper.CopyScheduleTimeToRetentionTimes((CmdletModel.VaultRetentionPolicy)retentionPolicy,
+                                                     (CmdletModel.SimpleSchedulePolicy)schedulePolicy);
+                    Logger.Instance.WriteDebug("Copy of RetentionTime from with SchedulePolicy to RetentionPolicy is successful");
+
+                    // Now validate both RetentionPolicy and SchedulePolicy together
+                    PolicyHelpers.ValidateLongTermRetentionPolicyWithSimpleSchedulePolicy(
+                                        (CmdletModel.VaultRetentionPolicy)retentionPolicy,
+                                        (CmdletModel.SimpleSchedulePolicy)schedulePolicy);
+                    Logger.Instance.WriteDebug("Validation of Retention policy with Schedule policy is successful");
+                    azureFileShareProtectionPolicy.RetentionPolicy = PolicyHelpers.GetServiceClientVaultRetentionPolicy(
+                                                   (CmdletModel.VaultRetentionPolicy)retentionPolicy);
+                }
+                else if (ProviderData.ContainsKey(PolicyParams.RetentionPolicy) && ProviderData[PolicyParams.RetentionPolicy] != null && ProviderData[PolicyParams.RetentionPolicy].GetType() == typeof(CmdletModel.LongTermRetentionPolicy))
+                {
+                    // validate RetentionPolicy
+                    AzureWorkloadProviderHelper.ValidateLongTermRetentionPolicy(retentionPolicy, BackupManagementType.AzureStorage, ScheduleRunFrequency);
+                    Logger.Instance.WriteDebug("Validation of Retention policy is successful");
+
+                    // update the retention times from backupSchedule to retentionPolicy after converting to UTC           
+                    AzureWorkloadProviderHelper.CopyScheduleTimeToRetentionTimes((CmdletModel.LongTermRetentionPolicy)retentionPolicy,
+                                                     (CmdletModel.SimpleSchedulePolicy)schedulePolicy);
+                    Logger.Instance.WriteDebug("Copy of RetentionTime from with SchedulePolicy to RetentionPolicy is successful");
+
+                    // Now validate both RetentionPolicy and SchedulePolicy together
+                    PolicyHelpers.ValidateLongTermRetentionPolicyWithSimpleSchedulePolicy(
+                                        (CmdletModel.LongTermRetentionPolicy)retentionPolicy,
+                                        (CmdletModel.SimpleSchedulePolicy)schedulePolicy);
+                    Logger.Instance.WriteDebug("Validation of Retention policy with Schedule policy is successful");
+                    azureFileShareProtectionPolicy.RetentionPolicy = PolicyHelpers.GetServiceClientLongTermRetentionPolicy(
+                                                    (CmdletModel.LongTermRetentionPolicy)retentionPolicy);
+                }
```

Comment:
this code might get reduced after we make LTR as base class for VaultRetentionPolicy

Reply:
Done